src/vs/workbench/services/notification/common/notificationService.ts

```diff
@@ -52,12 +57,63 @@ export class NotificationService extends Disposable implements INotificationServ
 	}
 
 	notify(notification: INotification): INotificationHandle {
-		return this.model.addNotification(notification);
+
+		let handle: INotificationHandle;
+		if (notification.neverShowOptions) {
+			const id = notification.neverShowOptions.id;
+			if (this.isRejected(id)) {
+				return new NoOpNotification();
+			}
+
+			const neverShowAction = new Action(
+				'workbench.dialog.choice.neverShowAgain',
+				nls.localize('neverShowAgain', "Don't Show Again"),
+				undefined, true, () => {
+					handle.close();
+					this.neverShow(id);
+					return Promise.resolve();
+				});
+
+			//default to secondary
+			if (typeof (notification.neverShowOptions.isSecondary) !== 'boolean') {
+				notification.neverShowOptions.isSecondary = true;
+			}
+			notification.actions = notification.actions || {};
+			if (notification.neverShowOptions.isSecondary) {
```

Comment:
I would suggest to add the "Don't show again" as last entry to the secondary actions.

Reply:
I am sorry I forgot what you said about it in your comment. Will fix.